src/main/java/com/iota/iri/service/API.java

```diff
@@ -417,13 +417,14 @@ private AbstractResponse checkConsistencyStatement(List<String> transactionsList
     }
 
     /**
-     * Compares the last received confirmed milestone with the last global snapshot milestone.
-     * If these are equal, it means the tangle is empty and therefore invalid.
+     * Checks whether the node is synchronized by comparing the latest known milestone index against the current
+     * latest solid milestone index. We allow for max. one milestone delta as a headroom, as a milestone is
+     * only applied in a set interval to the current ledger state.
      *
-     * @return <tt>false</tt> if we received at least a solid milestone, otherwise <tt>true</tt>
+     * @return true if the node is synced, false if not
      */
-    public boolean invalidSubtangleStatus() {
-        return (snapshotProvider.getLatestSnapshot().getIndex() == snapshotProvider.getInitialSnapshot().getIndex());
+    public boolean isNodeSynchronized() {
+        return snapshotProvider.getLatestSnapshot().getIndex() >= latestMilestoneTracker.getLatestMilestoneIndex() -1;
```

Comment:
Looking now more closely (not rushing like last time):
The original logic covered the case where a node attempted to sync from the global/local snapshot and no milestones were yet to be received. So it covered the case where a node had communication problems with the neighbors, or the short time it takes to receive the first milestone during gossip.

The idea originally was to check that the node got solid on at least one milestone. The new logic will return `true` in this case since in the beginning the `latestSnapshot` should be equal to `latestMilestoneIndex` (the global/local genesis milestone index).

So it is an edge case, but I think it should be covered.

So I suggest two things:
1. Combine the 2 conditions
```suggestion
        return (snapshotProvider.getLatestSnapshot().getIndex() != snapshotProvider.getInitialSnapshot().getIndex()) && (snapshotProvider.getLatestSnapshot().getIndex() >= latestMilestoneTracker.getLatestMilestoneIndex() -1);
```
2. Adding a unit test for this method testing the conditions:
- before milestones are received
- after milestones are received but not synced
- after milestones are received and synced

Reply:
Maybe don't commit the suggestion via github because it might create a long line :sweat_smile: